internal/provider/camunda_cluster_connector_secret_resource.go

```diff
@@ -0,0 +1,222 @@
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Manage a cluster connector secret on Camunda SaaS.",
+
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Cluster Connector Secret Name",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+				Validators: []validator.String{
+					validators.StringLengthBetweenValidator{Min: 1, Max: 50},
+					validators.StringNoSpacesValidator{},
+				},
+			},
+			"cluster_id": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Cluster ID",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"value": schema.StringAttribute{
+				MarkdownDescription: "The value of the connector secret",
+				Required:            true,
+				// Todo: Its actually also possible to update the secret value in-place. Not sure whether
+				// that just needs a different implementation or the API is not documented in the spec.
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+				Sensitive:     true,
+			},
+		},
+	}
+}
+
+func (r *CamundaClusterConnectorSecretResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Provider not yet configured
+	if req.ProviderData == nil {
+		return
+	}
+
+	provider, ok := req.ProviderData.(*CamundaCloudProvider)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *CamundaCloud client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.provider = provider
+}
+
+func (r *CamundaClusterConnectorSecretResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data camundaClusterConnectorSecret
+
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	newClusterConnectorSecretConfiguration := console.CreateSecretBody{
+		SecretName:  data.Name.ValueString(),
+		SecretValue: data.Value.ValueString(),
+	}
+
+	ctx = context.WithValue(ctx, console.ContextAccessToken, r.provider.accessToken)
+
+	inline, err := r.provider.client.ClustersApi.
+		CreateSecret(ctx, data.ClusterId.ValueString()).
+		CreateSecretBody(newClusterConnectorSecretConfiguration).
+		Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create cluster connector secret",
+			fmt.Sprintf("Unable to create cluster connector secret, got error: %s",
+				err.(*console.GenericOpenAPIError).Body()),
+		)
+		return
+	}
+
+	if inline.StatusCode >= 300 {
+		resp.Diagnostics.AddError(
+			"Unable to create cluster connector secret",
+			fmt.Sprintf("Unable to create cluster connector secret, got status: %s", inline.Status))
```

Comment:
`inline` is a bit of a weird variable name here perhaps?

```suggestion
	response, err := r.provider.client.ClustersApi.
		CreateSecret(ctx, data.ClusterId.ValueString()).
		CreateSecretBody(newClusterConnectorSecretConfiguration).
		Execute()

	if err != nil {
		resp.Diagnostics.AddError(
			"Unable to create cluster connector secret",
			fmt.Sprintf("Unable to create cluster connector secret, got error: %s",
				err.(*console.GenericOpenAPIError).Body()),
		)
		return
	}

	if response.StatusCode >= 300 {
		resp.Diagnostics.AddError(
			"Unable to create cluster connector secret",
			fmt.Sprintf("Unable to create cluster connector secret, got status: %s", response.Status))
```